packages/Ripple/src/index.js

```diff
@@ -0,0 +1,194 @@
+import Inferno   from 'inferno'
+import Component from 'inferno-component'
+import { bind }  from 'decko'
+
+export class Ripple extends Component {
+  state = {
+    ripples: []
+  }
+
+  componentDidMount() {
+    this.parent = this.ripple.parentElement
+
+    const { parent, handleMouseDown } = this
+    parent.addEventListener('mousedown', handleMouseDown)
+  }
+
+  componentWillUnmount() {
+    const { parent, handleMouseDown } = this
+    parent.removeEventListener('mousedown', handleMouseDown)
+  }
+
+  @bind
+  createListener(id) {
+    const { parent, startRemoval }  = this
+    const { ripples } = this.state
+
+    const handler = function() {
+      ripples[id].removable = true
+
+      // This will just work if the
+      // animation has ended
+      startRemoval(id)
+      parent.removeEventListener('mouseup', handler)
+    }
+
+    parent.addEventListener('mouseup', handler)
+  }
+
+  @bind
+  startRemoval(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+
+    if(!ripples[id].animationEnded || !ripples[id].removable)
+      return false
+
+    // If we can remove it we first start
+    // By decreasing the opacity to 0 and
+    // Then after the animation finishes we
+    // Remove the node
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+
+    setTimeout(e => this.removeRipple(id), opacityTiming || 300)
+  }
+
+  @bind
+  removeRipple(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+    setTimeout(() => {
+      ripples[id] = undefined
+      this.setState({ ripples })
+    }, 10000)
+  }
+
+  @bind
+  animationEnded(id) {
+    const { ripples } = this.state
+
+    ripples[id].animationEnded = true
+    ripples.removable = true
+    this.setState({ ripples })
+
+    if (ripples[id].removable) {
+      this.removeRipple
+    }
+
+    // This will just work if the user has
+    // fired the mouseup event
+    this.startRemoval(id)
+  }
+
+  @bind
+  startScaling(id) {
+    const { ripples } = this.state
+    const { scaleTiming } = this.props
+    const self = this
+
+    setTimeout(e => this.animationEnded(id), scaleTiming || 400)
+
+    const _styles = {
+      ...ripples[id].styles,
+      transform: 'scale(50)'
```

Comment:
The scale is too low. As you can see in the demo the ripple doesn't fill all the space of the containing div.
It should be at least 100 and should be made customizable for even bigger elements through the `scalingSize` attirbute/prop.